x-pack/legacy/plugins/maps/public/layers/styles/vector_style.js

```diff
@@ -448,14 +450,24 @@ export class VectorStyle extends AbstractStyle {
 
     const isDynamicConfigComplete = _.has(styleDescriptor, 'options.field.name')
       && _.has(styleDescriptor, 'options.color');
-    if (isDynamicConfigComplete) {
-      return this._getMBDataDrivenColor({
-        fieldName: styleDescriptor.options.field.name,
-        color: styleDescriptor.options.color,
-      });
+    if (!isDynamicConfigComplete) {
+      return null;
     }
 
-    return null;
+    if (styleDescriptor.options.useCustomColorRamp &&
+      (!styleDescriptor.options.customColorRamp ||
+      !styleDescriptor.options.customColorRamp.length)) {
```

Comment:
Can this `(!styleDescriptor.options.customColorRamp ||
      !styleDescriptor.options.customColorRamp.length)` ever arise? I think the UX already prevents creating dysfunctional color ramps. And if it doesn't, shouldn't it?

Reply:
> Can this (!styleDescriptor.options.customColorRamp || !styleDescriptor.options.customColorRamp.length) ever arise

Yes. The default vector style does not set  `customColorRamp` so when `useCustomColorRamp` is first set to true, `customColorRamp` is undefined until a color has been added with ColorStops